js-timezone-selector-plugin/src/selector/index.tsx

```diff
@@ -5,35 +5,39 @@ import Button from "./Button";
 import OptionsContainer from "./OptionsContainer";
 import { DEFAULT_VALUE } from './utils';
 
-const Selector = ({ className = "", position = "bottom", onChange = (selectedValue) => { } }) => {
+const Selector = ({ className = "", position = "bottom", onChange = (selectedValue) => { }, elementId = "timezone-selector" }) => {
   const [selectedValue, setSelectedValue] = useState(DEFAULT_VALUE);
   const [showOptions, setShowOptions] = useState(false);
   const top = position === "top";
 
-  const addOutsideClickListener = () => {
-    window.addEventListener('click', function (e) {
-      if (!(document.getElementById('timezoneSelector')?.contains(e.target))) {
-        setShowOptions(false)
+  useEffect(() => {
+    const addOutsideClickListener = (e) => {
+      if (!(document.getElementById(elementId)?.contains(e.target))) {
+        setShowOptions(false);
       }
-    });
-  };
+    };
 
+    window.addEventListener('click', addOutsideClickListener);
 
-  addOutsideClickListener();
+    return () => {
+      window.removeEventListener('click', addOutsideClickListener);
+    };
+  }, []);
 
   useEffect(() => {
     onChange(selectedValue);
   }, [selectedValue])
 
   return (
-    <div className={`flex flex-col relative w-full ${className}`} id="timezoneSelector">
-      <Button showOptions={showOptions} setShowOptions={setShowOptions} selectedValue={selectedValue} />
+    <div className={`flex flex-col relative w-full ${className}`} id={elementId}>
+      <Button showOptions={showOptions} setShowOptions={setShowOptions} selectedValue={selectedValue} elementId={elementId} />
```

Comment:
I believe you can use

```suggestion
      <Button {...{ elementId, selectedValue, isOverlayVisible, setIsOverlayVisible }} />
```